text/0000-mixed-shadow-mode.md

```diff
@@ -11,80 +11,88 @@ pr: https://github.com/salesforce/lwc-rfcs/pull/49
-that are not possible in a native Shadow DOM environment (e.g., instrumentation, styling, etc, which
-are discussed below).
+of these polyfills can result in broken experiences for existing applications due to dependencies on
+certain "escape hatches" that are not possible in a native Shadow DOM environment (e.g.,
+instrumentation, styling, etc).
 
 Applications that wish to use LWC components in a native shadow context simply need to omit the
-inclusion of the `@lwc/synthetic-shadow` polyfill; however, this is not a realistic option for
-large applications that cannot afford a rewrite.
+inclusion of the `@lwc/synthetic-shadow` polyfill; however, this all-or-nothing approach is not a
+realistic option for large applications that cannot afford a rewrite.
 
 This proposal opens up an incremental migration path for applications moving to native Shadow DOM by
-allowing the usage of both native and synthetic Shadow DOM in the same application.
+enabling the usage of both native and synthetic Shadow DOM in the same application.
 
 ## Detailed design
 
 Up until now, the two choices of shadow semantics have been synthetic mode and native mode. All the
 components in an application run in either synthetic mode or native mode, depending on whether the
 synthetic shadow polyfill (i.e., `@lwc/synthetic-shadow`) is applied. Mixed mode introduces the
-ability for a component to access native Shadow DOM APIs, even if the synthetic shadow polyfill is
-applied. Mixed mode should be transparent to components and the framework should guarantee this
+ability for a component to access native Shadow DOM APIs, even while the synthetic shadow polyfill
+is applied. Mixed mode should be transparent to components and the framework should guarantee this
 through integration testing, which is further discussed below.
 
 ### Opting in
 
-A component can signal that it prefers native Shadow DOM by setting a static property called
-`preferNativeShadow` to `true`.
+A component can signal that it supports both synthetic and native Shadow DOM by setting a static
+property called `shadowMode` to `any`.
 
 ```js
 export default class extends LightningElement {
-    static preferNativeShadow = true;
+    static shadowMode = 'any';
 }
 ```
 
-Using this property to opt in will result in the whole component subtree to operate in native shadow
-mode. and it is the responsibility of the component author to ensure that all components are
-compatible with native Shadow DOM. Due to the nature of this feature, it is expected that initial
-usage will start at the leaf components and work its way up the component tree.
+This mode will result in the whole component subtree to operate in either native shadow mode or
+synthetic shadow mode. If the browser supports Shadow DOM the shadow mode will be native. If the
+browser does not support Shadow DOM (e.g., IE11) the shadow mode will be synthetic. The test to
+determine Shadow DOM support will be to check whether `window.ShadowRoot` is defined.

```

Comment:
Bikeshed: The term `any` doesn't represent well the behavior described here. What do you think of: `prefer-native`, `native-if-available` ?

Reply:
@pmdartus It's hard to think of names that capture everything going on here. We have two modes:

1. Put me in native if the environment supports it, or if I'm off-core and there's no `@lwc/synthetic-shadow`
2. Put me in native if my ancestor forces me (by choosing option 1 above), or if I'm off-core and there's no `@lwc/synthetic-shadow`